app/api/teams.py

```diff
@@ -219,3 +221,219 @@ async def extend_team_trial(
         # Don't fail the request if email fails
 
     return {"message": "Team trial extended successfully"}
+
+def _check_key_name_conflicts(team1_keys: List[DBPrivateAIKey], team2_keys: List[DBPrivateAIKey]) -> List[str]:
+    """Return list of conflicting key names between two teams"""
+    team1_names = {key.name for key in team1_keys if key.name}
+    team2_names = {key.name for key in team2_keys if key.name}
+    return list(team1_names.intersection(team2_names))
+
+async def _resolve_key_conflicts(
+    conflicts: List[str],
+    strategy: str,
+    team2_keys: List[DBPrivateAIKey],
+    rename_suffix: Optional[str] = None,
+    db: Session = None,
+    current_user = None
+) -> List[DBPrivateAIKey]:
+    """Apply conflict resolution strategy to team2 keys"""
+    if strategy == "delete":
+        # Remove conflicting keys from team2 and delete them from database
+        keys_to_delete = [key for key in team2_keys if key.name in conflicts]
+        remaining_keys = [key for key in team2_keys if key.name not in conflicts]
+
+        # Delete conflicting keys from database if db session provided
+        if db and current_user:
+            for key in keys_to_delete:
+                try:
+                    await delete_private_ai_key(
+                        key_id=key.id,
+                        current_user=current_user,
+                        user_role="system_admin",  # System admin context for merge operations
+                        db=db
+                    )
+                except Exception as e:
+                    logger.error(f"Failed to delete key {key.id}: {str(e)}")
+                    # Continue with other keys even if one fails
+
+        return remaining_keys
+    elif strategy == "rename":
+        # Rename conflicting keys in team2
+        suffix = rename_suffix or "_merged"
+        for key in team2_keys:
+            if key.name in conflicts:
+                key.name = f"{key.name}{suffix}"
+        return team2_keys
+    elif strategy == "cancel":
+        # Return original keys unchanged
+        return team2_keys
+    else:
+        raise ValueError(f"Unknown conflict resolution strategy: {strategy}")
+
+async def _migrate_users_to_team(
+    users: List[DBUser],
+    target_team_id: int,
+    db: Session
+) -> int:
+    """Migrate users from source team to target team"""
+    migrated_count = 0
+    for user in users:
+        if user.team_id != target_team_id:
+            user.team_id = target_team_id
+            migrated_count += 1
+    db.commit()
```

Comment:
it might be fine but it seems like these interspersed commits could result in the DB getting into an inconsistent state if something fails partway through.